### PR TITLE
Remove ExtraCoverage category — run all functional tests in CI

### PR DESCRIPTION
**Pull request by tyrielv**
## Summary

The `ExtraCoverage` category excluded ~110 functional test methods (19 test classes) from the default test run and CI. These tests cover critical functionality - mount edge cases, dehydrate, repair, shared cache, disk layout upgrades, junctions, and more - but were never validated in CI, only via the `--extra-only` flag.

This PR removes ExtraCoverage filtering, fixes atrophied tests, and introduces infrastructure to track tests that still need work.

## Infrastructure changes

- **Remove ExtraCoverage filtering** - delete `ExtraCoverage` constant, `--extra-only` flag handling, and all `[Category(Categories.ExtraCoverage)]` annotations (19 usages)
- **Download FastFetch artifact** in `functional-tests.yaml` so FastFetch tests find the exe
- **Fix NUnitRunner slice grouping** - extend regex to include `MultiEnlistmentTests` alongside `EnlistmentPerFixture`, preventing Order-dependent tests from splitting across slices
- **Introduce `SkipInCIAttribute`** with a required reason string, replacing the bare `NeedsReactionInCI` category. Makes it clear why each test is skipped
- **Increase test slices** from 10 to 12 to distribute the additional test load
- **Resilient teardown** - `UnmountAndDeleteAll` catches `TimeoutException` from stuck unmounts and kills the GVFS.Mount process as a fallback, preventing a single hung unmount from eating 5 minutes of CI time
- **Update `AuthoringTests.md`** documentation

## Fixed tests (now running in CI)

| Test | Fix |
|------|-----|
| FastFetchTests (15 methods) | FastFetch artifact now downloaded in CI |
| ConfigVerbTests (8 methods) | Slice grouping keeps Order-dependent tests together |
| RepairTests (3 methods) | Removed stale mount-fail assertions - GVFS now tolerates corrupt index |
| FastFetchTests.CanFetchIntoEmptyGitRepoAndCheckoutWithGit | Fixed git output assertion (casing changed between git versions) |
| MountTests.MountFailsWhenNoOnDiskVersion | Wrapped in try/finally for reliable metadata restoration |
| MountTests.MountFailsWhenNoLocalCacheRootInRepoMetadata | Check exit code only (error messages now go to GVFS log, not stdout) |
| MountTests.MountFailsWhenNoGitObjectsRootInRepoMetadata | Same as above |
| MountTests.MountShouldFail helper | Now captures stderr in addition to stdout |
| All other MountTests | Fixed cascade failures via try/finally + stderr capture |

## Removed tests

| Test | Reason |
|------|--------|
| UpgradeReminderTests (3 methods) | Old NuGet-based upgrade reminder system removed from service |
| SharedCacheUpgradeTests (entire class) | Zero test methods - dead code |
| MountTests.MountMergesLocalPrePostHooksConfig | Mount no longer merges local hook config; just overwrites with GVFS.Hooks.exe |

## Still SkipInCI (follow-up work)

| Test | Count | Reason |
|------|------:|--------|
| DehydrateTests | 18 | Atrophied: folder dehydrate behavior changed substantially |
| WindowsDiskLayoutUpgradeTests | 4 | Atrophied: expected path/placeholder counts drifted |
| SharedCacheTests.RepairFixesCorruptBlobSizesDatabase | 1 | Product bug: mount crashes after repair of corrupt BlobSizes.sql |
| LooseObjectStepTests | 3 | Pre-existing flaky (timing-sensitive) |
| PrefetchVerbTests | 2 | Pre-existing flaky (blob count varies) |

## Timing impact

No measurable CI time increase. The additional ~80 tests that now run fit within the 12-slice parallelism. Baseline slowest slice was ~8.7 min; with the changes it's comparable.

### CODE REVIEW

File: GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs [hunk 1]
```diff
@@ -293,10 +293,49 @@ public string SetCacheServer(string arg)
...
+            }
+
             this.DeleteEnlistment();
         }
 
+        public void KillMountProcess()
+        {
+            try
+            {
+                foreach (var proc in System.Diagnostics.Process.GetProcessesByName("GVFS.Mount"))
```
Comment (KeithIsSleeping) [inline GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs]: On an unmount timeout, teardown kills every GVFS.Mount process on the machine. Because test fixtures run in parallel, this can also kill a healthy mount owned by another fixture running at the same time.

File: GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MountTests.cs [hunk 2]
```diff
@@ -202,14 +158,21 @@ public void MountFailsWhenNoLocalCacheRootInRepoMetadata()
...
+            {
+                this.fileSystem.CreateEmptyFile(metadataPath);
+                GVFSHelpers.SaveDiskLayoutVersion(this.Enlistment.DotGVFSRoot, majorVersion, minorVersion);
+                GVFSHelpers.SaveGitObjectsRoot(this.Enlistment.DotGVFSRoot, objectsRoot);
 
-            this.fileSystem.DeleteFile(metadataPath);
-            this.fileSystem.MoveFile(metadataBackupPath, metadataPath);
+                // Mount error messages go to the GVFS log, not stdout/stderr.
+                // Verify mount fails (exit code 3) without checking output text.
+                this.MountShouldFail(GVFSGenericError, expectedErrorMessage: null);
```
Comment (KeithIsSleeping) [inline GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MountTests.cs:167]: These metadata-corruption tests now check only that mount returns exit code 3, not why it failed, so a mount that fails for an unrelated reason would still pass.

File: GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs (hunk 1 above)
Comment (tyrielv) [reply GVFS/GVFS.FunctionalTests/Tools/GVFSFunctionalTestEnlistment.cs]: Fixed

File: GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MountTests.cs (hunk 2 above)
Comment (tyrielv) [reply GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/MountTests.cs:167]: Fixed

**Review by KeithIsSleeping — APPROVED** [review]

**Review by KeithIsSleeping — APPROVED** [review]
